EpsilonAuxBMS/Src/SetContactorsTask.c

```diff
@@ -47,344 +55,89 @@ void resetSenseArray(uint32_t senses[]);
 // Function for checking if CURRENT_SENSE is at a stable value
 int isSenseStable(uint32_t senses[]);
 
+// Functions for state's logic.
+void firstCheck(SetContactorStateInfo* stateInfo);
+void commonEnableCheck(SetContactorStateInfo* stateInfo);
+void chargeEnableCheck(SetContactorStateInfo* stateInfo);
+void dischargeEnableCheck(SetContactorStateInfo* stateInfo);
+void idle(SetContactorStateInfo* stateInfo);
+void blocked(SetContactorStateInfo* stateInfo);
+
+// Checks if stateInfo->hasDischargeBeenSet is set but discharge is disconnected
+const uint8_t dischargeDisconnectedDuringContactorSetting(const uint8_t dischargeSense, uint8_t hasDischargeBeenSet);
+
+// Checks if stateInfo->hasChargeBeenSet is set but charge disconnected
+const uint8_t chargeDisconnectedDuringContactorSetting(const uint8_t chargeSense, uint8_t hasChargeBeenSet);
+
+// Checks if discharge contactor can be turn on
+const uint8_t canTurnOnDishargeContactor(const uint8_t dischargeSense);
+
+// Checks if charge contactor can be turn on
+const uint8_t canTurnOnChargeContactor(const uint8_t chargeSense);
+
+// Checks if the discharge contactor should be on, but isn't
+const uint8_t dischargeDisconnectedUnexpectedly(const uint8_t dischargeSense, const uint8_t isdischargeTurningOn);
+
+// Checks if the charge contactor should be on, but isn't
+const uint8_t chargeDisconnectedUnexpectedly(const uint8_t chargeSense, const uint8_t isChargeTurningOn);
+
+//First check before setting states in case battery voltages are not in range, start up sequence is not done and makes sure the state is not disconnected
+const uint8_t safeToSetContactor(ContactorsSettingState state);
```

Comment:
Can we move these function declarations inside the header? That way unit tests can reference them later.